### ISSUE TITLE
CRI client-side implementation (aka kuberuntime) in kubelet

### ISSUE THREAD

**Issue by yujuhong** [kubernetes/kubernetes#31458]
Meta issue: #28789

Creating this issue to track the progress of kuberuntime.

- [ ] Support pod/container security context features (e.g., RunAsUser). 

Relevant PRs so far: #29884, #30121, #31322, #30267, #31320, #31395, #29811, #31326, #30753, #29811, #30212
Other relevant issues: #30463

**Comment by feiskyer** [comment]
@yujuhong It's better to make two list of those PRs: merged and on-reviewing ones.

**Comment by yujuhong** [comment]
@feiskyer do you have a list of missing features?

**Comment by yujuhong** [comment]
I just discovered that we don't support various security context features. For example, `RunAsUser` is never examined and applied to the container configs in kuberuntime.
I looked at the docker package and noticed that `DockerManager` calls methods in [`pkg/securitycontext`](https://github.com/kubernetes/kubernetes/blob/8fd414537b5143ab039cb910590237cabf4af783/pkg/securitycontext/types.go) directly to modify the docker configs. Obviously, this won't work for CRI. We'll need to change the package or implement similar functions in `kuberuntime`. The SecurityContext e2e tests are "feature" tests that don't run in regular e2e suites, and that's why we can't rely on e2e tests to catch the missing features.

/cc @pmorie @timstclair @feiskyer @euank @yifan-gu @Random-Liu
